### PR TITLE
Refactor Types, add new APITree and APIBody

### PR DESCRIPTION
**Pull request by durulkoca**

### CODE REVIEW

File: src/components/APIDialogAction/APIDialogAction.jsx [hunk 1]
```diff
@@ -1,14 +1,10 @@
...
-import Grid from "@mui/material/Grid";
 import SaveIcon from "@mui/icons-material/Save";
-import styles from "./styles";
-import { ToggleButton, ToggleButtonGroup } from "@mui/material";
-
+import { Box, Button, ToggleButton, ToggleButtonGroup } from "@mui/material";
 import { useEffect, useState } from "react";
 
-function APIDialogAction({
+export default function APIDialogAction({
```
Comment (canmingir) [inline src/components/APIDialogAction/APIDialogAction.jsx:7]: Please move all exports at the bottom

File: src/components/APIPath/APIPath.jsx [hunk 2]
```diff
@@ -1,166 +1,38 @@
...
+
+const APIPath = () => {
+  return (
+    <Grid container sx={styles.root}>
+      <Grid sx={styles.firstElement} />
+      <Grid item>
+        <Grid container item sx={styles.content}>
+          <Select defaultValue={""}>
+            {Constants.methods
+              .filter((methodName) => methodName)
```
Comment (canmingir) [inline src/components/APIPath/APIPath.jsx:15]: I don't think it does anything

**Review by canmingir — APPROVED** [review]

File: src/components/APIPath/APIPath.jsx (hunk 2 above)
Comment (durulkoca) [reply src/components/APIPath/APIPath.jsx]: i removed the line

File: src/components/APIDialogAction/APIDialogAction.jsx (hunk 1 above)
Comment (durulkoca) [reply src/components/APIDialogAction/APIDialogAction.jsx]: export moved to bottom

**Review by halilcengel — APPROVED** [review]

**Pull request merged**